egret/model_library/transmission/branch.py

```diff
@@ -689,3 +809,44 @@ def declare_ineq_angle_diff_branch_lbub(model, index_set,
             m.ineq_angle_diff_branch_ub[branch_name] = \
                 pe.atan(m.vj[from_bus] / m.vr[from_bus]) \
                 - pe.atan(m.vj[to_bus] / m.vr[to_bus]) <= math.radians(branches[branch_name]['angle_diff_max'])
+
+
+def declare_ineq_p_interface_lbub(model, index_set, interfaces):
+    """
+    Create the inequality constraints for the interface limits
+    based on the power variables or expressions.
+
+    p_interface_limits should be (lower, upper) tuple
+    """
+    m = model
+    con_set = decl.declare_set('_con_ineq_p_interface_lbub',
+                               model=model, index_set=index_set)
+
+    m.ineq_pf_interface_lb = pe.Constraint(con_set)
+    m.ineq_pf_interface_ub = pe.Constraint(con_set)
+
+    slacks_pos = hasattr(m, 'pfi_slack_pos')
+    slacks_neg = hasattr(m, 'pfi_slack_neg')
+    if slacks_pos != slacks_neg:
+        raise Exception("Both positive and negative interface slacks "\
+                "should be added to the transmission model")
+
+    for interface_name in con_set:
+        interface = interfaces[interface_name]
+
+        if interface['minimum_limit'] is not None:
+            expr_lower = m.pfi[interface_name]
+            if slacks_neg and interface_name in m.pfi_slack_neg:
+                expr_lower += m.pfi_slack_neg[interface_name]
+
+            m.ineq_pf_interface_lb[interface_name] = \
+                interface['minimum_limit'] <= expr_lower
+
+
+        if interface['maximum_limit'] is not None:
+            expr_upper = m.pfi[interface_name]
+            if slacks_pos and interface_name in m.pfi_slack_pos:
+                expr_upper -= m.pfi_slack_neg[interface_name]
```

Comment:
Shouldn't this be "m.pfi_slack_pos[interface_name]"?

Reply:
Yes, fixed! Thanks!